prefect_dbt/cloud/jobs.py

```diff
@@ -87,10 +143,174 @@ def trigger_job_run_flow():
         trigger_job_run_flow()
         ```
     """  # noqa
+    logger = get_run_logger()
+
+    logger.info(f"Triggering run for job with ID {job_id}")
+
     try:
         async with dbt_cloud_credentials.get_administrative_client() as client:
             response = await client.trigger_job_run(job_id=job_id, options=options)
     except HTTPStatusError as ex:
-        status = ex.response.json().get("status", {})
-        raise JobRunTriggerFailed(status.get("user_message")) from ex
-    return response
+        raise DbtCloudJobRunTriggerFailed(extract_user_message(ex)) from ex
+
+    run_data = response.json()["data"]
+
+    logger.info(
+        f"Run successfully triggered for job with ID {job_id}. "
+        "You can view the status of this run at "
+        f"https://{dbt_cloud_credentials.domain}/#/accounts/"
+        f"{dbt_cloud_credentials.account_id}/projects/{run_data['project_id']}/"
+        f"runs/{run_data['id']}/"
+    )
+
+    return run_data
+
+
+@task(
+    name="Get dbt Cloud job run details",
+    description="Retrieves details of a dbt Cloud job run "
+    "for the run with the given run_id.",
+    retries=3,
+    retry_delay_seconds=10,
+)
+async def get_dbt_cloud_run_info(
+    dbt_cloud_credentials: DbtCloudCredentials, run_id: int
+) -> Dict:
+    """
+    A task to retrieve information about a dbt Cloud job run.
+
+    Args:
+        dbt_cloud_credentials: Credentials for authenticating with dbt Cloud.
+        run_id: The ID of the job to trigger.
+
+    Returns:
+        The run data returned by the dbt Cloud administrative API.
+
+    Example:
+        Get status of a dbt Cloud job run:
+        ```python
+        from prefect import flow
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import get_run
+
+        @flow
+        def get_run_flow():
+            credentials = DbtCloudCredentials(api_key="my_api_key", account_id=123456789)
+
+            return get_run(
+                dbt_cloud_credentials=credentials,
+                run_id=42
+            )
+
+        get_run_flow()
+        ```
+    """  # noqa
+    try:
+        async with dbt_cloud_credentials.get_administrative_client() as client:
+            response = await client.get_run(run_id=run_id)
+    except HTTPStatusError as ex:
+        raise DbtCloudGetRunFailed(extract_user_message(ex)) from ex
+    return response.json()["data"]
+
+
+@flow
+async def trigger_dbt_cloud_job_run_and_wait_for_completion(
+    dbt_cloud_credentials: DbtCloudCredentials,
+    job_id: int,
+    trigger_job_run_options: Optional[TriggerJobRunOptions] = None,
```

Comment:
do you have an example of using `step_overrides` with this parameter?

Reply:
Added an example of how to include overrides when using this flow.

Reply:
Thank you! Looks splendid!